admin/reputation_simulator.php

```diff
@@ -0,0 +1,645 @@
+<?php
+/**
+ * @var System $system
+ * @var User $user
+ */
+require_once __DIR__ . "/_authenticate_admin.php";
+$self_link = $system->router->base_url . 'admin/reputation_simulator.php';
+$data = null;
+$debug_data = null;
+
+class RepPlayerReputation extends UserReputation {
+    public function runDecay(RepPlayer $repPlayer, $sim_data) {
```

Comment:
```suggestion
    public function runDecay(RepPlayer $repPlayer, SimDataDto $sim_data) {
    
    // ...
    
    class SimDataDto {
       public function __construct(
          public float $decay_modifier,
          public int $daily_pvp_cap,
          public int $pvp_daily_con,
          // ...
       ) {}
    }
```
Totally optional here, but I think things are more readable if we get out of the habit of using associative arrays as objects and use typed data classes instead.

The syntax in this comment is PHP short-form class initialization to take out some of the boilerplate of writing it, see here for details: https://stitcher.io/blog/constructor-promotion-in-php-8